graph.cpp

```diff
@@ -77,15 +111,22 @@ Supernode* Graph::generate_delta_node(uint64_t node_n, long node_seed, uint64_t
                             nondirectional_non_self_edge_pairing_fn(edge, src)));
     }
   }
-  return Supernode::delta_supernode(node_n, node_seed, updates);
+  return Supernode::delta_supernode(node_n, node_seed, updates, delta_loc);
 }
-void Graph::batch_update(uint64_t src, const std::vector<uint64_t>& edges) {
+void Graph::batch_update(uint64_t src, const std::vector<uint64_t>& edges, Supernode *delta_loc) {
   if (update_locked) throw UpdateLockedException();
+  // page in the supernodes
+  Supernode::page_in(delta_loc);
+  Supernode::page_in(supernodes[src]);
```

Comment:
unneeded?

Reply:
I'm not sure yet. I'll run a quick experiment and find out